src/include/common/dedicated_thread_owner.h

```diff
@@ -16,35 +18,65 @@ namespace terrier {
  */
 class DedicatedThreadOwner {
  public:
+  virtual ~DedicatedThreadOwner() = default;
   /**
    * @return the number of threads owned by this owner
    */
-  size_t GetThreadCount() { return thread_count_; }
+  size_t GetThreadCount() {
+    std::unique_lock<std::mutex> lock(thread_count_latch_);
+    return thread_count_;
+  }
+
+  /**
+   * Notifies the owner that a new thread can be given to it. The thread owner has the opportunity to decline the new
+   * thread if it does not need it
+   * @warning Should only be called by self driving infrastructure.
+   * @return true if owner accepts thread, false if it declines it
+   */
+  bool NotifyNewThread() { return OnThreadGranted(); }
```

Comment:
It's unclear what the difference between this and `GrantNewThread` is for the caller. If all the caller should do is call them in sequence, we might as well merge the two methods.